### ISSUE TITLE
[Request] Opltions to configure stuff like detect magic, tremorsense and or walls

### ISSUE THREAD

**Issue by maxobremer** [dev7355608/vision-5e#12]
It would be nice if in the options you could configure certain things in the option menu, like:

- What rarity of item is picked up by detect magic (personally I'd like it to be uncommon)
- To use the foundry version of tremorsense or this one
- To have any (or specific) detection/vision modes be blocked by walls or not
- configure which icon is seen when detecting with Detect Evil and Good, Detect Magic, Detect Poison and Disease, Detect Thoughts, Hearing, and Tremorsense (a poison icon for detect poison for example)

**Comment by GambetTV** [comment]
Just to piggy-back on this, I'd just like configuration options to turn on/off things in general. The Hearing thing is such a niche use-case that I basically can't see myself using that functionality ever, but can see a lot of potential headaches caused by not being able to disable it.

**Comment by LukasPrism** [comment]
I don’t disagree with giving people options but hearing isn’t niche. Invisible creatures should still be able to be located and targeted *unless* they’re hiding or also inaudible. So any time invisibility comes into play, hearing is very useful.

**Comment by GambetTV** [comment]
I see what you're saying, but unless I'm wrong there aren't specific rules around hearing, to the degree that it should be assumed that within 30 feet all characters should basically be capable of echolocating an invisible person's exact location by default. I definitely see why some people would want this feature, but for me I'd vastly prefer it shut off so that I can describe the scene instead of having everyone automatically know where everyone is. Also unless I'm wrong the 30 feet is pure homebrew. Which is all fine, I'm not criticizing, just saying I'd definitely appreciate some configuration options to turn this stuff on/off, or maybe configure some defaults around things like default Hearing range, etc.

**Comment by maxobremer** [comment]
tbh I would also like to either disable the hearing thing or make what the players see way less easy to see, like change the token opacity to 0.1, its too visible for being invisible

**Comment by LukasPrism** [comment]
Sorry to go on a bit of a tangent OP. It's perhaps relevant as it could be an option.

Eber, I'm not sure if I should ping you on the discord anymore (now that you're staff)? And I didn't want to create a new issue, as it may not indeed be an issue.

I've had this confirmed by another user, so it's not just my setup.

When using a Detection Mode, it totally overrides a token's vision – you are almost blind and would be stumbling into walls etc:
<img width="979" alt="Screenshot 2023-06-24 at 8 11 47 AM" src="https://github.com/dev7355608/vision-5e/assets/104886594/58d1d97a-8b24-40aa-9d4a-b2c973ec1ac7">

In fact it's worse than being blind where you can still see a faint background of explored canvas, bravo for that btw).

I would have expected Detection Modes to look more like this? (but with the detected element highlighted, I couldn't replicate that here)
<img width="758" alt="Screenshot_2023-06-24_at_8 18 32_AM" src="https://github.com/dev7355608/vision-5e/assets/104886594/82b61b3d-dfce-4468-bdaf-7b6e23e40876">

So you can still see what you're doing but are alerted to detected things. As it stands, say, for instance, you have Detect Poison going but stay on darkvision – if you can see the poisonous snake it doesn't highlight, but if you go around a corner it does. That doesn't seem right? You should know if it's poisonous if you can see it or not. 

Please feel free to correct me! Cheers

**Comment by LukasPrism** [comment]
> I see what you're saying, but unless I'm wrong there aren't specific rules around hearing, to the degree that it should be assumed that within 30 feet all characters should basically be capable of echolocating an invisible person's exact location by default. I definitely see why some people would want this feature, but for me I'd vastly prefer it shut off so that I can describe the scene instead of having everyone automatically know where everyone is. Also unless I'm wrong the 30 feet is pure homebrew. Which is all fine, I'm not criticizing, just saying I'd definitely appreciate some configuration options to turn this stuff on/off, or maybe configure some defaults around things like default Hearing range, etc.

PS. I can see your point – options would be great. The hearing thing could be contentious. There may already be a sneaky way to disable it but it would be great to know.

**Comment by LukasPrism** [comment]
> tbh I would also like to either disable the hearing thing or make what the players see way less easy to see, like change the token opacity to 0.1, its too visible for being invisible

The opacity thing is a core feature if I'm not mistaken. I'm sure this module could provide a solution but it's not really its responsibility. I wonder if you use a Custom Convenient Effect whether you can override the core behaviour for that 🤔

**Comment by dev7355608** [comment]
### Item Rarity

Item rarity is a mess in the dnd5e compendium packs. There are also people that use rarity for nonmagical items, which makes it impossible for Detect Magic to work for those users. So I think the system needs to add a `magical` property to items, otherwise it's impossible to make Detect Magic work for everybody.

### Default Hearing Range

The default hearing range is now configurable. I added a setting for it.

### Tremorsene / Detect \*

If I didn't make the scene completely black, then tokens with "Detect \*" would be able to see things they shouldn't be able to see. Detect Magic should only show magical auras, but parts of the scene, especially if it's behind a wall, which this sense ignore.

https://github.com/dev7355608/vision-5e/releases/tag/v1.4.0

---

@LukasPrism You can ping on Discord. I just might forget to respond if I can't respond immediately.

**Comment by dev7355608** [comment]
@maxobremer

> To have any (or specific) detection/vision modes be blocked by walls or not

Are there specific detection modes you'd like to have ignore walls or not ignore walls? What's the use case exactly?

> configure which icon is seen when detecting with Detect Evil and Good, Detect Magic, Detect Poison and Disease, Detect Thoughts, Hearing, and Tremorsense (a poison icon for detect poison for example)

That wouldn't be so easy to implement because core only returns the first detection mode that is able to detect the target token. It doesn't give you a full report of which senses detect the token. What you describe would be preferable, but, I think, it would require even more invasive changes to core, which I'm trying to avoid at the moment.

**Comment by LukasPrism** [comment]
> ### Item Rarity
> Item rarity is a mess in the dnd5e compendium packs. There are also people that use rarity for nonmagical items, which makes it impossible for Detect Magic to work for those users. So I think the system needs to add a `magical` property to items, otherwise it's impossible to make Detect Magic work for everybody.
> 
> ### Default Hearing Range
> The default hearing range is now configurable. I added a setting for it.
> 
> ### Tremorsene / Detect *
> If I didn't make the scene completely black, then tokens with "Detect *" would be able to see things they shouldn't be able to see. Detect Magic should only show magical auras, but parts of the scene, especially if it's behind a wall, which this sense ignore.
> 
> https://github.com/dev7355608/vision-5e/releases/tag/v1.4.0
> 
> @LukasPrism You can ping on Discord. I just might forget to respond if I can't respond immediately.

Hey, thanks so much for the detailed response! And *love* the new update. I wonder if there's a way we could see creatures at <0 elevation still? Maybe you could solve this v10 problem of the tokens disappearing behind the canvas? Now that we have burrowing status, they could only disappear _if_ burrowing? Then things like canyons, pits on maps will mean we can have negative elevation but still see the tokens (_unless_ they're burrowing).

Thanks for clearing up the intent with the black screen on the detection modes. I get it now, and can liken it to closing your eyes and focusing on the detecting part. I think I was confused because I mistakenly thought you *had* to switch to detect things, but that isn't the case. 

But I still have a conceptual issue with the new detection modes. Unlike tremorsense, echolocation etc, they're more than just providing the fact that an entity exists – they also provide important information about that token (if it is magical, fey etc). At the moment this information is only conveyed if you *can't* see the token:
<img width="713" alt="Screenshot 2023-06-26 at 12 46 28 PM" src="https://github.com/dev7355608/vision-5e/assets/104886594/3d9c8da3-b211-485c-934f-224e1db8b6da">

You have to move out of sightline to know if something is picked up by the spell/ability:
<img width="721" alt="Screenshot 2023-06-26 at 12 47 46 PM" src="https://github.com/dev7355608/vision-5e/assets/104886594/753a8889-0a2f-4f5c-9c6d-a315f612272e">

Is this something that can be fixed? Even if just as an option?

Many thanks again for this wonderful module!

**Comment by LukasPrism** [comment]
Oh and totally agree with the Item Rarity thing – I really hope dnd5e fix that ASAP. I wonder if in the interim your module could ignore creatures that have 'Common' on their attack items? That will fix a lot of the false positives for detect magic picking up SRD creatures incorrectly.

**Comment by MestreMahakala** [comment]
I would like to be able to add hearing through formulas so that I can make the passive perception of creatures a determining factor in the values. Or at least a flag in the MID? Thank you.

**Comment by dev7355608** [comment]
- Detect Magic/... now always apply their filter to the token. I hope I didn't break anything: the patching of core has become quite the mess.
- Moved the background image to the background.

https://github.com/dev7355608/vision-5e/releases/tag/v1.5.0

---

@LukasPrism I haven't made that change yet, but I think I have to treat common items as nonmagical in general.

**Comment by maxobremer** [comment]
> @maxobremer
> 
> > To have any (or specific) detection/vision modes be blocked by walls or not
> 
> Are there specific detection modes you'd like to have ignore walls or not ignore walls? What's the use case exactly?
> 


So regarding walls, its more that most maps I personally use the walls would definitely block magic, due to their thickness and material. By having it block it would make a lot more sense. Like most dungeons id say have walls that block magic, cause 1 foot isnt much and thats all it needs for stone. that applies to all detect X spells except for detect thoughts btw. thought detect thoughts can also be blocked by thick enough walls, so maybe have that one be blocked by a terrain wall, since those often are quit thick.

**Comment by maxobremer** [comment]
> ### Item Rarity
> Item rarity is a mess in the dnd5e compendium packs. There are also people that use rarity for nonmagical items, which makes it impossible for Detect Magic to work for those users. So I think the system needs to add a `magical` property to items, otherwise it's impossible to make Detect Magic work for everybody.
> 

To get it perfect for everyone is never gonna work, though by making it configurable as a setting, what rarity is picked up by detect magic, then people can at least make it to their liking. cause like you said cant really get it right without a magic property. For my own games just increasing it from detecting common and up to uncommon and up, would solve like 95% of situations where i see detect magic having an issue in my games.

**Comment by LukasPrism** [comment]
> * Detect Magic/... now always apply their filter to the token. I hope I didn't break anything: the patching of core has become quite the mess.
> * Moved the background image to the background.
> 
> https://github.com/dev7355608/vision-5e/releases/tag/v1.5.0
> 
> @LukasPrism I haven't made that change yet, but I think I have to treat common items as nonmagical in general.

Thanks so much for the changes you have made in the latest update. The detection modes are working wonderfully now and you fixed one of my biggest annoyances of v10 with the canvas elevation.

You’ve made 5E *so* much better… bravo.

**Comment by dev7355608** [comment]
> > @maxobremer
> > > To have any (or specific) detection/vision modes be blocked by walls or not
> > 
> > 
> > Are there specific detection modes you'd like to have ignore walls or not ignore walls? What's the use case exactly?
> 
> So regarding walls, its more that most maps I personally use the walls would definitely block magic, due to their thickness and material. By having it block it would make a lot more sense. Like most dungeons id say have walls that block magic, cause 1 foot isnt much and thats all it needs for stone. that applies to all detect X spells except for detect thoughts btw. thought detect thoughts can also be blocked by thick enough walls, so maybe have that one be blocked by a terrain wall, since those often are quit thick.

@maxobremer I decided to have Detect Magic/... be restricted by sight-blocking walls. It might take the fun out of it in some cases not being able to detect something in an adjacent room, but it's probably for the better overall. I thought about using sound-blocking instead of sight-blocking walls: the idea is that, if a wall lets sound through it, must be relatively thin and therefore let magic pass. But that's probably too weird. If anyone has thoughts on this, let me know.

https://github.com/dev7355608/vision-5e/releases/tag/v1.6.0

**Comment by LukasPrism** [comment]
Hey thanks for fixing the background tile elevation thing, works wonderfully now!

Since this is the thread for asking for options – I'm going to post this here. We had our first session in AHD with V5E last night and it works a treat. 

My only question is if there's a way we can possibly modify how the detected creatures are visualised? As you can see from this screen – the hearing detected token REALLY jumps out in the gloomy setting:

<img width="900" alt="Screenshot 2023-07-03 at 2 39 25 PM" src="https://github.com/dev7355608/vision-5e/assets/104886594/1c279b65-23d5-4591-8046-d6b1cb1a72e9">

This is a bit of a mood-killer and quite distracting. I actually really like the soundwave animation behind the mystery man and wonder if we could just have an option for that to be a little bit stronger and no icon?

Food for thought... thanks!

**Comment by gambit07** [comment]
> Hey thanks for fixing the background tile elevation thing, works wonderfully now!
> 
> Since this is the thread for asking for options – I'm going to post this here. We had our first session in AHD with V5E last night and it works a treat.
> 
> My only question is if there's a way we can possibly modify how the detected creatures are visualised? As you can see from this screen – the hearing detected token REALLY jumps out in the gloomy setting:
> <img alt="Screenshot 2023-07-03 at 2 39 25 PM" width="900" src="https://user-images.githubusercontent.com/104886594/250430536-1c279b65-23d5-4591-8046-d6b1cb1a72e9.png">
> 
> This is a bit of a mood-killer and quite distracting. I actually really like the soundwave animation behind the mystery man and wonder if we could just have an option for that to be a little bit stronger and no icon?
> 
> Food for thought... thanks!

Seconding this, would love a more subdued option here!

**Comment by dev7355608** [comment]
I'm closing this issue. Please open new issues for requests that haven't been addressed yet. One issue per feature request please.
